src/Identifier/LdapIdentifier.php

```diff
@@ -0,0 +1,216 @@
+     *
+     * @var \Authentication\Identifier\Ldap\AdapterInterface
+     */
+    protected $_ldap = null;
+
+    /**
+     * {@inheritDoc}
+     */
+    public function __construct(array $config = [])
+    {
+        parent::__construct($config);
+
+        $this->_checkLdapConfig();
+        $this->_buildLdapObject();
+    }
+
+    /**
+     * Checks the LDAP config
+     *
+     * @throws \RuntimeException
+     * @throws \InvalidArgumentException
+     * @return void
+     */
+    protected function _checkLdapConfig()
+    {
+        if (!isset($this->_config['bindDN'])) {
+            throw new RuntimeException('Config `bindDN` is not set.');
+        }
+        if (!is_callable($this->_config['bindDN'])) {
+            throw new InvalidArgumentException(sprintf(
+                'The `bindDN` config is not a callable. Got `%s` instead.',
+                gettype($this->_config['bindDN'])
+            ));
+        }
+        if (!isset($this->_config['host'])) {
+            throw new RuntimeException('Config `host` is not set.');
+        }
+    }
+
+    /**
+     * Constructs the LDAP object and sets it to the property
+     *
+     * @throws \RuntimeException
+     * @return void
+     */
+    protected function _buildLdapObject()
+    {
+        $ldap = $this->_config['ldap'];
+
+        if (is_string($ldap)) {
+            $class = App::className($ldap, 'Identifier/Ldap');
+            $ldap = new $class();
+        }
+
+        if (!($ldap instanceof AdapterInterface)) {
+            throw new RuntimeException('Option `ldap` must implement Authentication\Identifier\Backend\LdapInterface.');
+        }
+
+        $this->_ldap = $ldap;
+    }
+
+    /**
+     * Identify
+     *
+     * @param array $data Authentication credentials
+     * @return \Cake\Datasource\EntityInterface|null
+     */
+    public function identify($data)
+    {
+        $this->_connectLdap();
+        $fields = $this->getConfig('fields');
+
+        if (isset($data[$fields['username']]) && isset($data[$fields['password']])) {
+            return $this->_bindUser($data[$fields['username']], $data[$fields['password']]);
+        }
+
+        return null;
+    }
+
+    /**
+     * Initializes the LDAP connection
+     *
+     * @return void
+     * @throws \Cake\Network\Exception\InternalErrorException Raised in case of an unsucessful connection.
+     */
+    protected function _connectLdap()
+    {
+        $config = $this->getConfig();
+
+        $this->_ldap->connect(
+            $config['host'],
+            $config['port'],
+            $this->getConfig('options')
+        );
+    }
+
+    /**
+     * Try to bind the given user to the LDAP server
+     *
+     * @param string $username The username
+     * @param string $password The password
+     * @return \Cake\Datasource\EntityInterface|null
+     */
+    protected function _bindUser($username, $password)
+    {
+        // Turn all LDAP errors into exceptions
+        set_error_handler(
```

Comment:
I guess we have to wrap each ldap_* call in a new error handler then, as `set_error_handler` is applied globally?